File: mobile/android/app/src/main/java/com/whereswaldo/android/ui/designsystem/token/TypographyTokens.kt
package com.whereswaldo.android.ui.designsystem.token

import androidx.compose.runtime.Immutable
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

/** The six normative type roles (specs/003-android-client.md §4.1/§4.2). Same values in light
 * and dark — only [com.whereswaldo.android.ui.designsystem.token.WaldoColorTokens] varies by
 * theme. */
@Immutable
data class WaldoTypographyTokens(
    val displayLarge: TextStyle,
    val titleLarge: TextStyle,
    val titleMedium: TextStyle,
    val bodyLarge: TextStyle,
    val bodyMedium: TextStyle,
    val labelSmall: TextStyle,
)

// Waldo design system (design/waldo-design-system/) — platform system font (Roboto); sizes in sp.
val WaldoTypography = WaldoTypographyTokens(
    displayLarge = TextStyle(fontSize = 34.sp, lineHeight = 40.sp, fontWeight = FontWeight.Bold, letterSpacing = (-0.68).sp),
    titleLarge = TextStyle(fontSize = 22.sp, lineHeight = 28.sp, fontWeight = FontWeight.SemiBold, letterSpacing = (-0.22).sp),
    titleMedium = TextStyle(fontSize = 17.sp, lineHeight = 22.sp, fontWeight = FontWeight.SemiBold),
    bodyLarge = TextStyle(fontSize = 17.sp, lineHeight = 24.sp, fontWeight = FontWeight.Normal),
    bodyMedium = TextStyle(fontSize = 15.sp, lineHeight = 20.sp, fontWeight = FontWeight.Normal),
    labelSmall = TextStyle(fontSize = 12.sp, lineHeight = 16.sp, fontWeight = FontWeight.Medium, letterSpacing = 0.4.sp),
)

val LocalWaldoTypography = staticCompositionLocalOf { WaldoTypography }

File: mobile/ios/WaldoKit/Sources/WaldoKit/DesignSystem/Tokens/TypographyTokens.swift
import SwiftUI

/// specs/004-ios-client.md §2.1 — the six type roles, identical across light/dark (typography
/// doesn't change with color scheme, but is exposed via `Theme` regardless so a future design pass
/// can still override it uniformly).
public struct TypographyTokens: Equatable {
    public var displayLarge: Font
    public var titleLarge: Font
    public var titleMedium: Font
    public var bodyLarge: Font
    public var bodyMedium: Font
    public var labelSmall: Font

    public init(displayLarge: Font, titleLarge: Font, titleMedium: Font, bodyLarge: Font, bodyMedium: Font, labelSmall: Font) {
        self.displayLarge = displayLarge
        self.titleLarge = titleLarge
        self.titleMedium = titleMedium
        self.bodyLarge = bodyLarge
        self.bodyMedium = bodyMedium
        self.labelSmall = labelSmall
    }

    // Waldo design system (design/waldo-design-system/) — SF Pro system font. `titleLarge` is
    // Semibold (was Bold) per the unified type scale. Line-height/tracking (design README) are
    // applied by components at render; the token contract's type is `Font`.
    public static let standard = TypographyTokens(
        displayLarge: .system(size: 34, weight: .bold),
        titleLarge: .system(size: 22, weight: .semibold),
        titleMedium: .system(size: 17, weight: .semibold),
        bodyLarge: .system(size: 17, weight: .regular),
        bodyMedium: .system(size: 15, weight: .regular),
        labelSmall: .system(size: 12, weight: .medium)
    )
}
